src/status_im/data_store/realm/core.cljs

```diff
@@ -20,18 +20,18 @@
         (aset arr i (aget key i)))
       (.-buffer arr))))
 
-(defn realm-version
+(defn unencrypted-realm?
+  [file-name]
+  (boolean
+   (.schemaVersion rn-dependencies/realm file-name)))
+
+(defn encrypted-realm-version
   [file-name encryption-key]
-  (if encryption-key
-    ;; we need to try this if previous version was using unencrypted database
-    (try
-      (.schemaVersion rn-dependencies/realm file-name (to-buffer encryption-key))
-      (catch js/Object e
-        (log/info "Attempting to read encrypted file failes")))
-    (try
-      (.schemaVersion rn-dependencies/realm file-name)
-      (catch js/Object e
-        (log/info "Attempting to read unencrypted file failed")))))
+  ;; we need to try this if previous version was using unencrypted database
+  (try
+    (.schemaVersion rn-dependencies/realm file-name (to-buffer encryption-key))
+    (catch js/Object e
+      (log/info "Attempting to read encrypted file failes"))))
```

Comment:
Can we improve error handling? What if this fails ?
(I realize you didn't introduce this)

Reply:
In this case it's ok for us to fail, we want to detect unencrypted versions of realm, so we can reset them, once we remove the code I agree error handling should be made more robust.

Reply:
What about returning nil only and not logging then?

Reply:
you mean just remove the log statement? (we need the schema version if it we manage to decrypt it)

Reply:
Right in the catch

Reply:
Hopefully this function does not only return nil :sweat_smile: